developers/transformations/index.md

```diff
@@ -0,0 +1,195 @@
+
+A `TriggerProfile` makes it possible to link a `TriggerChannel` to an `Item`.
+This `Profile` receives the `State` of the `Item` and the `Event` that has been triggered.
+
+## ProfileTypeProvider
+
+Custom `ProfileType`s have to be annouced by a `ProfileTypeProvider` to the framework via an OSGi service:
+
+```java
+@Component(service = { ProfileTypeProvider.class })
+public class MyProfileTypeProvider implements ProfileTypeProvider {
+    @Override
+    public Collection<ProfileType> getProfileTypes(Locale locale) {
+        //return custom types
+    }
+}
+```
+
+## ProfileFactory
+
+The most important interface is the `ProfileFactory` which has to be implemented and announced as an OSGi service:
+
+```java
+@Component(service = { ProfileFactory.class })
+public class MyProfileFactory implements ProfileFactory {
+```
+
+Such a factory is responsible for specific `ProfileTypeUID`s that should be returned by `Collection<ProfileTypeUID> getSupportedProfileTypeUIDs()`.
+Further it is capable of creating these `Profile`s via `createProfile(ProfileTypeUID profileTypeUID, ProfileCallback callback, ProfileContext profileContext);`.
+
+For convenience, the `ProfileFactory` and the `ProfileTypeProvider` can be put into one class and announce the two services:
+
+```java
+@Component(service = { ProfileFactory.class, ProfileTypeProvider.class })
+public class MyProfileFactory implements ProfileFactory, ProfileTypeProvider {
+```
+
+### ProfileCallback
+
+`Profile`s need the opportunity to notify the framework about what they did with the `Command`s and `StateUpdate`s it received from the framework.
+The `ProfileCallback` provides methods to forward the results of a `Profile`s processing to the `ThingHandler` and to the `Framework`.
+It should be injected into the `Profile` upon its creation.
+
+### ProfileContext
+
+Some more advanced `Profile`s which can be configured need access to their `Configuration` object.
+This is offered via the `ProfileContext`.
+A `ScheduledExecutorService` can also be retrieved via the `ProfileContext` in order to schedule long running tasks in a separate Thread.
+The `ProfileContext` may also be injected into the `Profile` upon its creation.
+
+## ProfileAdvisor
+
+A `ProfileAdvisor` is an optional component which can be used to suggest a specific `ProfileTypeUID` for a given `Channel` or `ChannelType`.
+Two methods have to be implemented to achieve this:
+
+`ProfileTypeUID getSuggestedProfileTypeUID(Channel channel, @Nullable String itemType);`
+
+`ProfileTypeUID getSuggestedProfileTypeUID(ChannelType channelType, @Nullable String itemType);`
+
+## Using Profiles
+
+### .items file
+
+`Profiles`s can be specified as a parameter for a given channel:
+
+```java
+<item-type> MyItem { channel="<bindingID>:<thing-typeID>:MyThing:myChannel"[profile="MyScope:MyProfile"]}
+```
+
+## Existing Profiles
+
+A few profiles are provided by openHAB itself.
```

Comment:
Maybe already document the new timestamp profiles here ( https://github.com/openhab/openhab-core/pull/732 )

Reply:
We do this in a follow up pr